packages/labs/virtualizer/src/test/scenarios/range-changed-events.test.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {
+  array,
+  ignoreBenignErrors,
+  isInViewport,
+  last,
+  until,
+} from '../helpers.js';
+import {LitVirtualizer} from '../../lit-virtualizer.js';
+import {RangeChangedEvent} from '../../Virtualizer.js';
+import {expect, html, fixture} from '@open-wc/testing';
+
+describe('RangeChanged event', () => {
+  ignoreBenignErrors(beforeEach, afterEach);
+
+  async function createVirtualizer<T>(properties: {items: T[]}) {
+    const container = await fixture(html` <div>
+      <style>
+        lit-virtualizer {
+          height: 200px;
+          width: 200px;
+        }
+        .item {
+          width: 200px;
+          height: 50px;
+          margin: 0;
+          padding: 0;
+        }
+      </style>
+      <lit-virtualizer
+        scroller
+        .items=${properties.items}
+        .renderItem=${(item: T) => html`<div class="item">${item}</div>`}
+      ></lit-virtualizer>
+    </div>`);
+    const virtualizer = (await until(() =>
+      container.querySelector('lit-virtualizer')
+    )) as LitVirtualizer;
+    expect(virtualizer).to.be.instanceof(LitVirtualizer);
+    return {container, virtualizer};
+  }
+
+  function getVisibleItems(virtualizer: LitVirtualizer) {
+    return Array.from(virtualizer.renderRoot.querySelectorAll('.item')).filter(
+      (e) => isInViewport(e, virtualizer)
+    ) as HTMLElement[];
+  }
+
+  it('should fire when rendered items have changed', async () => {
+    const items = array(1000);
+    const {container, virtualizer} = await createVirtualizer({items});
+    const containerEvents: RangeChangedEvent[] = [];
+    const virtualizerEvents: RangeChangedEvent[] = [];
+
+    await until(() => getVisibleItems(virtualizer).length === 4);
+
+    container.addEventListener('rangeChanged', (e) => {
+      containerEvents.push(e as RangeChangedEvent);
+    });
+
+    virtualizer.addEventListener('rangeChanged', (e) => {
+      virtualizerEvents.push(e as RangeChangedEvent);
+    });
+
+    virtualizer.scrollToIndex(500);
+
+    await until(() => virtualizerEvents.length === 1);
+
+    expect(last(virtualizerEvents).first).to.be.closeTo(480, 20); // +/- 20
```

Comment:
In this case I actually didn't run into variability.  The motivation for using closeTo in the tests for this was to express that the "overhang" number was intended to be more of a fuzzy heuristic thing than a specific value.  I also could have rebuilt the function to produce the expected overhang count in the test, but I thought that might be weird and heavy.  When/if we extend the virtualizer API to tune the out-of-view/overhang range explicitly, then it would make sense to be more definite with the value in the test.

Reply:
Sigh, given that overhang is currently locked in at 1000... I suppose I can just make this an explicit 20 for now.